app/controllers/estates_controller.rb

```diff
@@ -12,12 +12,15 @@ def new
 
   def create  
     @estate = @user.estates.new(estate_params)
-    if @estate.save
-      flash[:notice] = "Estate created successfully!"
-      redirect_to user_estates_path
-    else
-      flash.now.alert = "Estate not Created"
-      render :new
+    respond_to do |format|
+      if @estate.save
+        flash[:notice] = "Estate created successfully!"
+        format.html { redirect_to user_estates_path }
+      else
+        flash.now.alert = "Estate not Created"
+        format.html { render :new }
+        format.js
```

Comment:
Why is the js format is only in else condition?

Reply:
It is handled only when estate is not created